### ISSUE TITLE
Maximum call stack size exceeded on unavailable memcached

### ISSUE THREAD

**Issue by OndroNR** [animir/node-rate-limiter-flexible#33]
When memcached is unavailable, I get "Maximum call stack size exceeded" Error on line https://github.com/animir/node-rate-limiter-flexible/blob/master/lib/RateLimiterMemcache.js#L31
called from https://github.com/animir/node-rate-limiter-flexible/blob/master/lib/RateLimiterMemcache.js#L59 .

`attemptNumber` is never increased, which results in never ending loop. I believe `attemptNumber` should be increased on this line https://github.com/animir/node-rate-limiter-flexible/blob/master/lib/RateLimiterMemcache.js#L57

Reproduction code, memcached must be unavailable
```js
'use strict';

const {RateLimiterMemcache} = require('rate-limiter-flexible');
const Memcached = require('memcached');

const memcacheRateLimiterOpts = {
    points: 100, // 100 same events
    duration: 3600, // per 1 hour
    storeClient: new Memcached('127.0.0.1:12345', {timeout: 500, failures: 2, retries: 2})
};

const rateLimiter = new RateLimiterMemcache(memcacheRateLimiterOpts);

async function foobar() {
    await rateLimiter.consume('aaa');
}

foobar();
```

**Comment by animir** [comment]
@OndroNR Good catch.
PR is welcomed, are you going to create it?

**Comment by OndroNR** [comment]
@animir I can create it, but I am not sure about mocking unavailable memcached in tests. `memcached-mock` does not appear to support failure cases. Maybe just using something just like this?

```js
const mockClient = {
add: function(key, value, lifetime, cb) {cb(new Error("Server unavailable"));},
incr: function(key, amount, cb) {cb(new Error("Server unavailable"));}
}
```

**Comment by animir** [comment]
@OndroNR nice!
yeah, I agree with how you mocked unavailable server

**Comment by animir** [comment]
@OndroNR I've published version `1.0.3` with the fix you made

**Comment by OndroNR** [comment]
@animir thanks for fast release :)
